### ISSUE TITLE
Attr vs item storage

### ISSUE THREAD

**Issue by nsheff** [pepkit/attmap#50]
updating the attmap doesn't update the `to_yaml` function:

```
In [45]: import attmap                                                                                                             

In [46]: mt = attmap.PathExAttMap()                                                                                                

In [47]: mt.attr = "value"                                                                                                         

In [48]: mt.attr                                                                                                                   
Out[48]: 'value'

In [49]: mt.to_yaml()                                                                                                              
Out[49]: '{}'

In [50]: mt.to_map()                                                                                                               
Out[50]: OrderedDict([('attr', 'value')])

In [51]: mt.to_dict()                                                                                                              
Out[51]: {'attr': 'value'}
```

```
In [55]: mt = attmap.PathExAttMap({"orig": "orig_value"})                                                                          

In [56]: mt.attr = "value"                                                                                                         

In [57]: mt.attr                                                                                                                   
Out[57]: 'value'

In [58]: mt.to_yaml()                                                                                                              
Out[58]: 'orig: orig_value\nattr: value'

In [59]: mt.orig = "new_value"                                                                                                     

In [60]: mt.to_yaml()                                                                                                              
Out[60]: 'orig: orig_value\nattr: value'
```

**Comment by nsheff** [comment]
actually to_dict is also not updating it seems:

```
In [68]: mt.to_dict()                                                                                                              
Out[68]: {'attr': 'value', 'orig': 'orig_value'}
```

this is very strange I can't figure out what it's doing...

**Comment by nsheff** [comment]
ok this looks like a pretty deep rooted problem that's not restricted to `to_yaml`. I'm going to hold off on doing more with updates for now because it looks like the update mechanisms are broken. I wonder if this is related to the recent looper bug?

Here I'm trying to assign a new value to an attribute, and it shows up when I then try to access the attribute, but not when I try to print it or save it to yaml:

```In [70]: rgc = refgenconf.RefGenomeConfiguration("/ext/yeti/genomes//test2.yaml")                                                  

In [71]: rgc.genomes                                                                                                               

In [73]: hasattr(rgc, "genomes")                                                                                                   
Out[73]: True

In [74]: not rgc.genomes                                                                                                           
Out[74]: True

In [75]: rgc.genomes = attmap.PathExAttMap()                                                                                       

In [76]: rgc.genomes                                                                                                               
Out[76]: PathExAttMap: {}

In [77]: rgc                                                                                                                       
Out[77]: 
RefGenomeConfiguration
genome_server: http://localhost
genome_folder: /ext/yeti/genomes
genomes: null

In [78]: print(rgc)                                                                                                                
RefGenomeConfiguration
genome_server: http://localhost
genome_folder: /ext/yeti/genomes
genomes: null

In [79]: rgc.genomes = attmap.PathExAttMap({"hg38": 1})                                                                            

In [80]: rgc                                                                                                                       
Out[80]: 
RefGenomeConfiguration
genome_server: http://localhost
genome_folder: /ext/yeti/genomes
genomes: null

In [81]: rgc.genomes                                                                                                               
Out[81]: 
PathExAttMap
hg38: 1

In [82]: rgc                                                                                                                       
Out[82]: 
RefGenomeConfiguration
genome_server: http://localhost
genome_folder: /ext/yeti/genomes
genomes: null

In [83]: print(rgc)                                                                                                                
RefGenomeConfiguration
genome_server: http://localhost
genome_folder: /ext/yeti/genomes
genomes: null

In [84]: attmap.__version__                                                                                                        
Out[84]: '0.11'
```

**Comment by nsheff** [comment]
sorry, just realized I'm 2 hours behind on release! I"ll try it on 0.12.

**Comment by nsheff** [comment]
Ok, I can confirm all these problems still on 0.12.

**Comment by vreuter** [comment]
Will be fixed in 0.12.1 that's in progress. It's because item/attr storage differs

**Comment by vreuter** [comment]
The most basic type that should suffer this problem is `OrdAttMap`

**Comment by vreuter** [comment]
This seems to be resolved in 0.12.1

**Comment by nsheff** [comment]
:fireworks:
